sklearn/base.py

```diff
@@ -119,6 +119,12 @@ class BaseEstimator:
     arguments (no ``*args`` or ``**kwargs``).
     """
 
+    def __new__(cls, *args, **kwargs):
+        est = super().__new__(cls)
+        if cls.__module__.startswith("sklearn."):
```

Comment:
You should safegaurd against not having `__module__`, not all classes have them. I'm not sure if that can ever be the case for something inheriting from `BaseEstimator`, but while working in `skops` we discovered many cases where `__module__` is not there, or is modified to something which is not true (like, a scipy ufunc pretending it belongs to numpy).